src/v/archival/ntp_archiver_service.cc

```diff
@@ -753,6 +758,15 @@ ss::future<bool> ntp_archiver::maybe_upload_manifest(const char* upload_ctx) {
 }
 
 ss::future<> ntp_archiver::maybe_flush_manifest_clean_offset() {
+    if (
+      _projected_manifest_clean_at.has_value()
+      && ss::lowres_clock::now() - _last_manifest_upload_time
```

Comment:
Did you mean `_last_marked_clean_time` instead of `_last_manifest_upload_time`? Using `_last_manifest_upload_time` is also a good heuristic.